File: ChatGPT/c/output/output_06a.c
// ChatGPT-generated solution will go here.
#include <stdio.h>
#include <string.h>

#define MAX_SIZE 1000

int directions[4][2] = {
    {-1, 0}, // up
    {0, 1},  // right
    {1, 0},  // down
    {0, -1}  // left
};

char grid[MAX_SIZE][MAX_SIZE];
int visited[MAX_SIZE][MAX_SIZE] = {0};

int main() {
    FILE *f = fopen("input.txt", "r");
    char line[MAX_SIZE];
    int x = 0, y = 0, dir = 0;

    // Read the grid
    for (int i = 0; fgets(line, sizeof(line), f); i++) {
        strcpy(grid[i], line);
        for (int j = 0; j < strlen(line); j++) {
            if (line[j] == '^') {
                x = i;
                y = j;
                dir = 0; // Up
                grid[i][j] = '.';
            }
        }
    }

    fclose(f);

    int visited_count = 0;
    while (x >= 0 && x < MAX_SIZE && y >= 0 && y < MAX_SIZE && grid[x][y] != '\0') {
        if (!visited[x][y]) {
            visited[x][y] = 1;
            visited_count++;
        }

        int next_x = x + directions[dir][0];
        int next_y = y + directions[dir][1];

        if (grid[next_x][next_y] == '#') {
            dir = (dir + 1) % 4; // Turn right (clockwise)
        } else {
            x = next_x;
            y = next_y;
        }
    }

    printf("%d\n", visited_count);
    return 0;
}

File: ChatGPT/python/output/output_06a.py
# ChatGPT-generated solution will go here.
def main():
    with open('input.txt') as f:
        grid = [list(line.strip()) for line in f if line.strip()]

    rows, cols = len(grid), len(grid[0])

    directions = {
        '^': (-1, 0),
        '>': (0, 1),
        'v': (1, 0),
        '<': (0, -1),
    }
    turn_right = {
        '^': '>',
        '>': 'v',
        'v': '<',
        '<': '^',
    }

    # Find starting position and direction
    for r in range(rows):
        for c in range(cols):
            if grid[r][c] in directions:
                pos = (r, c)
                dir = grid[r][c]
                break

    visited = set()
    visited.add(pos)

    while True:
        dr, dc = directions[dir]
        nr, nc = pos[0] + dr, pos[1] + dc
        if 0 <= nr < rows and 0 <= nc < cols and grid[nr][nc] != '#':
            pos = (nr, nc)
            visited.add(pos)
        else:
            dir = turn_right[dir]
        # Exit condition: out of bounds in the next move
        dr, dc = directions[dir]
        next_r, next_c = pos[0] + dr, pos[1] + dc
        if not (0 <= next_r < rows and 0 <= next_c < cols):
            break
    print(len(visited))

if __name__ == '__main__':
    main()
